tests/chainer_tests/functions_tests/array_tests/test_get_item.py

```diff
@@ -76,6 +76,111 @@ def test_backward_gpu(self):
                             cuda.to_gpu(self.gy_data))
 
 
+@testing.parameterize(*testing.product_dict(
+    [
+        {'slices': [1, 0], 'sliced_shape': (2, 3, 2)},
+        {'slices': ([1, 0],), 'sliced_shape': (2, 3, 2)},
+        {'slices': numpy.array([[1, 0], [2, 3]]),
+         'sliced_shape': (2, 2, 3, 2)},
+        {'slices': ([1, 0], [1, 1]), 'sliced_shape': (2, 2)},
+        {'slices': ([1, 0], slice(None), [[1, 1], [1, 1]]),
+         'sliced_shape': (2, 2, 3)},
+        {'slices': ([1, 0], slice(1, 2), [0, 0]), 'sliced_shape': (2, 1)},
+        {'slices': ([[1, 1], [1, 0]], slice(1, 2), 1),
+         'sliced_shape': (2, 2, 1)},
+        {'slices': numpy.array([True] * 18 + [False] * 6).reshape(4, 3, 2),
+         'sliced_shape': (18,)},
+        {'slices': numpy.array([True, False, False, True]),
+         'sliced_shape': (2, 3, 2)},
+        {'slices': (slice(None), numpy.array([True, False, True])),
+         'sliced_shape': (4, 2, 2)},
+    ],
```

Comment:
Thanks for pointing this out.
I found that `cupy.scatter_add` can not handle that kind of input. Could you review this PR after that fix is merged?